aten/src/ATen/native/CPUBlas.cpp

```diff
@@ -220,6 +228,41 @@ void gemm(
       transa, transb, m, n, k, alpha, a, lda, b, ldb, beta, c, ldc);
 }
 
+void gemm(
+   TransposeType transa, TransposeType transb,
+   int64_t m, int64_t n, int64_t k,
+   const decltype(c10::impl::ScalarTypeToCPPType<at::kBFloat16>::t) alpha,
+   const decltype(c10::impl::ScalarTypeToCPPType<at::kBFloat16>::t) *a, int64_t lda,
+   const decltype(c10::impl::ScalarTypeToCPPType<at::kBFloat16>::t) *b, int64_t ldb,
+   const decltype(c10::impl::ScalarTypeToCPPType<at::kBFloat16>::t) beta,
+   decltype(c10::impl::ScalarTypeToCPPType<at::kBFloat16>::t) *c, int64_t ldc) {
+   internal::normalize_last_dims(transa, transb, m, n, k, &lda, &ldb, &ldc);
+#if AT_BUILD_WITH_BLAS() && defined(BLAS_HAS_SBGEMM)
+   if (use_blas_gemm(transa, transb, m, n, k, lda, ldb, ldc)) {
+      int m_ = m, n_ = n, k_ = k, lda_ = lda, ldb_ = ldb, ldc_ = ldc;
+      char transa_ = to_blas(transa), transb_ = to_blas(transb);
+      // alpha and beta and C matrix in OpenBLAS sbgemm are of type "float" so we have to convert, copy and copy back.
+      float alpha_ = (float) alpha, beta_ = (float) beta;
+      int c_size = n_ * ldc_;
+      float *c_float = (float *) malloc(c_size * sizeof(float));
+      for (int i=0; i<c_size; ++i) c_float[i] = (float) c[i];
+      sbgemm_(&transa_, &transb_,
+              &m_, &n_, &k_,
+              &alpha_,
+              a, &lda_,
+              b, &ldb_,
+              &beta_,
+              c_float, &ldc_);
+      for (int i=0; i<c_size; ++i) c[i] = (decltype(c10::impl::ScalarTypeToCPPType<at::kBFloat16>::t)) c_float[i];
+      free(c_float);
```

Comment:
Please consider replacing explicit loop with range loop or `std::transofrm(float_v.begin(), float_v.end(), c, [](float v) -> _bfloat16_t { return v;});`
```suggestion
              float_v.data(), &ldc_);
      for (auto cv: float_v) {
        *(c++) = static_cast<_bfloat16_t>(cv);
      }
```